### ISSUE TITLE
Implement better string <-> number conversion functions

### ISSUE THREAD

**Issue by gw280** [flutter/flutter#78802]
Right now in a11y we have a very rudimentary StringUtils collection that contains a number of methods such as `NumberToString`.

These need to be made a little bit better by:

- Formatting strings more appropriately when converting from numbers. e.g. `(int)6` becomes `"6.000000"` right now, but it should be `"6"`.
- Implementing `StringToInt` etc that do a better job (e.g. by correctly signalling errors, which `std::stoi` doesn't do) and using them where we would need to validate that the input string can actually be represented as a number.

**Comment by pedromassangocode** [comment]
Hi @gw280 
From what I can see, the issue is related to Dart rather than to Flutter itself. Please open the issue in the dedicated [repository](https://github.com/dart-lang/sdk/issues).
Closing, as this isn't an issue with Flutter itself.  
If you disagree, please write in the comments,  
providing your `flutter doctor -v`, your `flutter run -v`,  your `pubspec.yaml`
a minimal reproducible code sample that does not use 3rd party plugins,  
and I will reopen it.
Thank you

**Comment by gw280** [comment]
This is flutter engine - https://github.com/flutter/engine/blob/master/third_party/accessibility/base/string_utils.cc is the current code, we're planning to unfork this via the FML in https://github.com/flutter/flutter/issues/75653 and implement better string methods

**Comment by cbracken** [comment]
The fixes for this have all landed.
